secapi/observers.go

```diff
@@ -0,0 +1,34 @@
+package secapi
+
+import (
+	"time"
+)
+
+type resourceStateObserver[V comparable, R any] struct {
+	delay       time.Duration
+	interval    time.Duration
+	maxAttempts int
+	actFunc     func() (V, *R, error)
+}
+
+func (retry *resourceStateObserver[V, R]) WaitUntil(expectedValue V) (*R, error) {
```

Comment:
I recommend to use the https://pkg.go.dev/github.com/cenkalti/backoff/v4 package for this